documentation/_posts/2016-06-07-process-api.md

```diff
@@ -6,13 +6,53 @@ categories:
 weight: 6
 ---
 
-The main idea behind process api is having all port stuff events into one place, and all of the outputs sent out from the same place.
+First of all let's see what a component defined using Process API looks like:
 
-The way the process api works is it gets called for each event.
-If `done` does not get called, it keeps getting called, and the IPs that are passed to it keep getting appended to the buffer.
+```coffeescript
+exports.getComponent = ->
+  new noflo.Component
+    description: "Forward received data"
+    inPorts:
+      in:
+        datatype: 'all'
+        description: 'An input port'
+    outPorts:
+      out:
+        datatype: 'all'
+        description: 'An output port'
+    process: (input, output) ->
+      return unless input.has 'in'
+      data = input.getData 'in'
+      output.send data
+      output.done()
+```
+
+Note the `process: (input, output) ->`, that's the main method on
+Process API and gives its name. The `input` is a reference to all
+incoming IPs to the component and `output` is a reference to all
+outgoing IPs from the component.
+
+<!-- I missed an introduction that answers my questions: "What is the main
+goal/benefit of using Process API? Why I should not use other thing or just
+vanilla Components? Do I have more control? Does it helps with async
+components? What about error handling? I think here is where we "sell" the
+idea of Process API to people -->
```

Comment:
Yes it is naturally async, we don't show any other kind of components in the documentation except in legacy so it is meant to be the only way to do it.